src/canister/individual_user_template/src/api/cycle_management/return_cycles_to_user_index_canister.rs

```diff
@@ -20,16 +19,15 @@ async fn return_cycles_to_user_index_canister(cycle_amount: Option<u128>) {
             .unwrap()
     });
 
-    if api_caller != user_index_canister_id {
-        return;
+    
+    if cycle_amount.is_some() || canister_balance128() > INDIVIDUAL_USER_CANISTER_RECHARGE_AMOUNT {
```

Comment:
should we also check for cycle_amount > canister_balance before depositing ?

Reply:
then the call would fail on it's own. it's good to have let the call fail in that case and return the error.